client/order/index.js

```diff
@@ -97,4 +111,124 @@ jQuery( function ( $ ) {
 		document.body.appendChild( container );
 		ReactDOM.render( modalToRender, container );
 	}
+
+	function maybeShowDisputeNotice() {
+		const container = document.querySelector(
+			'#wcpay-order-payment-details-container'
+		);
+
+		// If the container doesn't exist (WC < 7.9), or the charge ID isn't present, don't render the notice.
+		if ( ! container || ! chargeId ) {
+			return;
+		}
+
+		ReactDOM.render( <DisputeNotice chargeId={ chargeId } />, container );
+	}
 } );
+
+const DisputeNotice = ( { chargeId } ) => {
```

Comment:
The logic below is almost identical to before the changes in aab6e0c where I've moved this to a separate `<DisputeNotice />` component to improve readability.

The data from `useCharge` is almost identical to what was previously provided by PHP in `disputeData`, so not many changes were required.